src/components/CActorArgument.vue

```diff
@@ -0,0 +1,296 @@
+        <label
+          class="appearance-none block w-full h-full mb-4 bg-gray-200 text-gray-700 border border-gray-200 rounded py-3 px-4 leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
+        >
+          {{ index }} : {{ el }}
+        </label>
+        <button
+          type="button"
+          @click="$delete(value, index, 1)"
+          class="bg-red-500 text-white font-bold py-2 px-4 rounded h-full flex-shrink-0 w-32"
+        >
+          Remove
+        </button>
+      </div>
+    </div>
+    <VueCtkDateTimePicker
+      v-else-if="type === 'datetime'"
+      v-model="value"
+      format="YYYY-MM-DD HH:mm"
+    />
+    <VueCtkDateTimePicker
+      v-else-if="type === 'date'"
+      v-model="value"
+      formatted="ll"
+      only-date
+      format="YYYY-MM-DD"
+    />
+    <textarea
+      v-else
+      class="block appearance-none w-full mb-4 bg-gray-200 border border-gray-200 text-gray-700 py-3 px-4 pr-8 rounded leading-tight focus:outline-none focus:bg-white focus:border-gray-500"
+      v-model="value"
+    />
+  </div>
+</template>
+
+<script>
+import draggable from 'vuedraggable';
+import utils from '@/utils';
+
+export default {
+  name: 'CActorArgument',
+  props: {
+    name: String,
+    type: String
+  },
+  components: {
+    draggable
+  },
+  data: function() {
+    let value;
+    let listType = '';
+    let typedValue = '';
+    let keyValue = '';
+    switch (this.type) {
+      case 'int':
+      case 'float':
+        value = 0;
+        break;
+      case 'bool':
+        value = false;
+        break;
+      case 'dict':
+        value = {};
+        break;
+      case 'list':
+        listType = 'text';
+        value = [];
+        break;
+      default:
+        value = '';
+    }
+
+    let keyType = '';
+    let valueType = '';
+
+    if (this.type.match('^List')) {
+      value = [];
+      listType = 'text';
+      if (this.type === 'List[bool]') {
+        listType = 'checkbox';
+        typedValue = false;
+      } else if (this.type === 'List[int]' || this.type === 'List[float]') {
+        listType = 'number';
+      }
+    }
+    if (this.type.match(/^Dict/)) {
+      value = {};
+      keyType = this.type.substring(this.type.search(/\[/) + 1, this.type.search(','));
+      if (keyType === 'bool') {
+        keyValue = false;
+      }
+      valueType = this.type.substring(this.type.search(',') + 2, this.type.length - 1);
+      if (valueType === 'bool') {
+        typedValue = false;
+      }
+    }
+    this.$emit('input', value);
+    return {
+      value: value,
+      typedKeyValue: keyValue,
+      typedValue: typedValue,
+      listType: listType,
+      keyType: keyType,
+      valueType: valueType
+    };
+  },
+  watch: {
+    value: function(newValue) {
+      this.$emit('input', newValue);
+    }
+  },
+
+  methods: {
+    addElement: function() {
+      if (this.type.match(/List\[typing.Dict/) || this.type.match(/List\[typing.List/)) {
```

Comment:
The goal here is to make sure make sure the element to be added is parsed as JSON and prevent it from being written as string in case the argument is a List of Lists or of Dicts. There was a missing 'else' a few lines below but it does seem to work.